collector/src/main.rs

```diff
@@ -1141,7 +1164,99 @@ fn main_result() -> anyhow::Result<i32> {
 
             Ok(0)
         }
+        Commands::Download(opts) => {
+            let crate_path = match opts.command {
+                DownloadSubcommand::Git { url } => {
+                    download_from_git(&benchmark_dir, opts.name.as_deref(), &url)?
+                }
+                DownloadSubcommand::Crate { krate, version } => {
+                    download_from_crates_io(&benchmark_dir, &krate, &version, opts.name.as_deref())?
+                }
+            };
+            prepare_downloaded_benchmark(&crate_path);
+            Ok(0)
+        }
+    }
+}
+
+fn download_from_git(
+    benchmark_dir: &Path,
+    name: Option<&str>,
+    url: &str,
+) -> anyhow::Result<PathBuf> {
+    let target = PathBuf::from(benchmark_dir);
+    let crate_name = match name {
+        Some(name) => name,
+        None => url
+            .trim_end_matches("/")
+            .trim_end_matches(".git")
+            .split("/")
+            .last()
+            .expect("Crate name could not be determined from git URL"),
+    };
+    let target = target.join(crate_name);
+    if target.exists() {
+        return Err(anyhow::anyhow!(
+            "Directory {} already exists",
+            target.display()
+        ));
+    }
+
+    Command::new("git")
+        .arg("clone")
```

Comment:
I suppose that this could use `--depth 1`, but it's probably not very important and it puts unnecessary strain on GitHub anyway.